src/methods/clipping/intersection.jl

```diff
@@ -0,0 +1,244 @@
+        end
+    end
+
+    # If the polygons don't intersect and aren't contained within each
+    # other, return_polys will be empty
+    return return_polys
+end
+
+"""
+    _get_inter_holes(return_polys, poly_a, poly_b)::Vector{Vector{Vector{Tuple{Float64, Float64}}}}
+
+When the _trace_difference function was called, it only took into account the
+exteriors of the two polygons when computing the difference. The function
+'_get_difference_holes' takes into account the holes of the original polygons
+and adjust the output of _trace_difference (return_polys) accordingly.
+
+## Example
+
+TODO
+```
+"""
+
+function _get_inter_holes(return_polys, poly_a, poly_b)
+    # Initiaze our return object
+    final_polys =  Vector{Vector{Vector{Tuple{Float64, Float64}}}}(undef, 0)
+
+    for poly in return_polys
+        # Turning polygon into the desired return type I can add more polygons to it
+        poly = [[poly]]
+
+        # We subtract the holes of 'poly_a' and 'poly_b' from the output we got
+        # from _trace_intersection (return_polys)
+        for hole in GI.gethole(poly_a) 
+            replacement_p = Vector{Vector{Vector{Tuple{Float64, Float64}}}}(undef, 0)
+            for p in poly
+                # When we take the difference of our existing intersectio npolygons and 
+                # the holes of polygon_a, we might split it up into smaller polygons. 
+                new_ps = difference(GI.Polygon(p), _lin_ring_to_poly(hole))
+                append!(replacement_p, new_ps)
+            end
+            poly = replacement_p
+        end
+        
+        for hole in GI.gethole(poly_b)
+            replacement_p = Vector{Vector{Vector{Tuple{Float64, Float64}}}}(undef, 0)
+            for p in poly
+                # When we take the difference of our existing intersectio npolygons and 
+                # the holes of polygon_a, we might split it up into smaller polygons. 
+                new_ps = difference(GI.Polygon(p), _lin_ring_to_poly(hole))
+                append!(replacement_p, new_ps)
+            end
+            poly = replacement_p
+        end
+        
+        append!(final_polys, poly)
+    end
+
+    return final_polys
+        
+end
+
+"""
+    intersection(
+        ::GI.PolygonTrait, poly_a,
+        ::GI.PolygonTrait, poly_b,
+    )::Vector{Vector{Vector{Tuple{Float64}}}}    
+
+Calculates the intersection between two polygons. If the intersection is empty, 
+the vector of a vector is empty (note the outermost vector is technically not empty).
+## Example
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+p1 = GI.Polygon([[(0.0, 0.0), (5.0, 5.0), (10.0, 0.0), (5.0, -5.0), (0.0, 0.0)]])
+p2 = GI.Polygon([[(3.0, 0.0), (8.0, 5.0), (13.0, 0.0), (8.0, -5.0), (3.0, 0.0)]])
+GO.intersection(p1, p2)
+
+# output
+1-element Vector{Vector{Vector{Tuple{Float64, Float64}}}}:
+[[[(6.5, 3.5), (10.0, 0.0), (6.5, -3.5), (3.0, 0.0), (6.5, 3.5)]]]
+```
+"""
+intersection(geom_a, geom_b) =
```

Comment:
Same commentary as difference. These boys go to the top since they are most important. You only need examples for functions that users are going to use. Remove TODOs and if the issues/questions still exist then open an issue